File: docs/docs/API-Reference/javascript-examples/api-flows-run/pass-global-variables-in-request-headers.js
const url = `${process.env.FLOW_SERVER_URL ?? ""}/api/v1/run/${process.env.FLOW_ID ?? ""}`;

const options = {
  method: 'POST',
  headers: {
    "Content-Type": `application/json`,
    "x-api-key": `${process.env.FLOW_API_KEY ?? ""}`,
    "X-FLOW-GLOBAL-VAR-OPENAI_API_KEY": `sk-...`,
    "X-FLOW-GLOBAL-VAR-USER_ID": `user123`,
    "X-FLOW-GLOBAL-VAR-ENVIRONMENT": `production`,
  },
  body: JSON.stringify({
  "input_value": "Tell me about something interesting!",
  "input_type": "chat",
  "output_type": "chat"
}),
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));

File: docs/docs/API-Reference/python-examples/api-flows-run/pass-global-variables-in-request-headers.py
import os

import requests

base = os.environ.get("FLOW_URL") or os.environ.get("FLOW_SERVER_URL", "")
flow_id = os.environ.get("FLOW_ID", "")
api_key = os.environ.get("FLOW_API_KEY", "")

headers = {
    "Content-Type": "application/json",
    "x-api-key": api_key,
    "X-FLOW-GLOBAL-VAR-USER_ID": "user123",
    "X-FLOW-GLOBAL-VAR-ENVIRONMENT": "production",
}

payload = {
    "input_value": "Tell me about something interesting!",
    "input_type": "chat",
    "output_type": "chat",
}

response = requests.post(f"{base}/api/v1/run/{flow_id}", headers=headers, json=payload, timeout=60)
response.raise_for_status()
print(response.text)
